internal/controller/management_controller.go

```diff
@@ -147,20 +151,47 @@ func (r *ManagementReconciler) update(ctx context.Context, management *kcm.Manag
 	}
 
 	if err := r.ensureAccessManagement(ctx, management); err != nil {
+		record.Warnf(management, nil, "EnsureAccessManagementFailed", "failed to ensure AccessManagement is created: %v", err)
 		l.Error(err, "failed to ensure AccessManagement is created")
 		return ctrl.Result{}, err
 	}
 
 	if err := r.enableAdditionalComponents(ctx, management); err != nil { // TODO (zerospiel): i wonder, do we need to reflect these changes and changes from the `wrappedComponents` in the spec?
+		record.Warnf(management, nil, "EnableAdditionalComponentsFailed", "failed to enable additional components: %v", err)
 		l.Error(err, "failed to enable additional KCM components")
 		return ctrl.Result{}, err
 	}
 
-	components, err := getWrappedComponents(management, release)
+	requeue, errs := r.reconcileManagementComponents(ctx, management, release)
+
+	shouldRequeue, err := r.startDependentControllers(ctx, management)
 	if err != nil {
-		l.Error(err, "failed to wrap KCM components")
+		record.Warnf(management, nil, "ControllersStartFailed", "Failed to start dependent controllers: %v", err)
 		return ctrl.Result{}, err
 	}
+	if shouldRequeue {
+		requeue = true
+	}
+
+	r.setReadyCondition(management)
+
+	if err := r.Client.Status().Update(ctx, management); err != nil {
+		errs = errors.Join(errs, fmt.Errorf("failed to update status for Management %s: %w", management.Name, err))
+	}
+
+	if errs != nil {
+		l.Error(errs, "Multiple errors during Management reconciliation")
+		return ctrl.Result{}, errs
+	}
```

Comment:
```suggestion
	errs = errors.Join(errs, r.updateStatus(ctx, management))
	if errs != nil {
		l.Error(errs, "Multiple errors during Management reconciliation")
		return ctrl.Result{}, errs
	}
```

Should we revert to the previous lines?